jimmy/modules/node_envvars/tests/test_node_envvars.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+
+#    Copyright 2017 Evgeny Sharypin
+#
+#    This program is free software; you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation; either version 2 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License along
+#    with this program; if not, write to the Free Software Foundation, Inc.,
+#    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import mock
+import mockfs
+import os
+import sys
+import jsonschema
+import pytest
+from jimmy import cli
+from click.testing import CliRunner
+from jimmy.lib.common import yaml_reader
+from jimmy.tests import base
+
+modules_dir = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
+jimmy_dir = os.path.join(os.path.dirname(modules_dir))
+jenkins_schema_path = os.path.join(modules_dir, 'node_envvars', 'resources', 'schema.yaml')
+jenkins_yaml_path = os.path.join(jimmy_dir, 'sample', 'input', 'jenkins.yaml')
+
+
+class TestNodeEnvVarsConfiguration(base.TestCase):
+
+    def setup_method(self, method):
+        self.runner = CliRunner()
+
+    def teardown_method(self, method):
+        mockfs.restore_builtins()
+
+    @mock.patch('jimmy.lib.core.load_py_modules')
+    @mock.patch('subprocess.call')
+    def test_cli_call(self, mock_subp, mock_modules):
+        with open(jenkins_schema_path, 'r') as f:
+            mock_jenkins_schema = f.read()
+        self.mfs = mockfs.replace_builtins()
+        self.mfs.add_entries({os.path.join(jimmy_dir, 'lib', 'schema.yaml'): self.jimmy_schema,
+                              os.path.join(jimmy_dir, 'jimmy.yaml'): self.mock_jimmy_yaml,
+                              jenkins_schema_path: mock_jenkins_schema,
+                              jenkins_yaml_path: '\n'.join(
+                                  [
+                                      'jenkins:',
+                                      '  slave-envvars:',
```

Comment:
Something is incorrect here.. You provided example with "node-env-vars", but here "slave-envvars" is used. This is probably because of too permissive schema. 
BTW IMHO "node-env-vars" sounds better.